### PR TITLE
LT-1525: Clearance application final advice permission (and bugs: 2011, 2064)

### PR DESCRIPTION
**Pull request by stephanieboyd**
- Add new permission for giving final advice on clearance applications

- Rename old final advice permission

- Fix bug 2064 and 2111

### CODE REVIEW

File: cases/views/advice.py [hunk 1]
```diff
@@ -354,7 +357,7 @@ def post(self, request, *args, **kwargs):
         if res.status_code == 400:
             form = approve_licence_form(
                 case_id=case["id"],
-                standard=standard,
+                is_open_licence=standard,
```
Comment (Mark-Spiers-kainos) [inline cases/views/advice.py]: can you double check this? It looks like it assigning if it's standard currently

File: cases/views_helpers.py [hunk 2]
```diff
@@ -42,19 +48,6 @@ def get_case_advice(get_advice, request, case, user_team_final, team=None):
...
-    for item in advice["advice"]:
-        if item["type"]["key"] == "conflicting":
-            able_to_finalize = False
-            break
-
-    able_to_create_and_edit_advice = "MANAGE_TEAM_CONFIRM_OWN_ADVICE" in permissions or (
-        "MANAGE_TEAM_ADVICE" in permissions and not case.get("has_advice").get("my_user")
-    )
-
-    context["able_to_finalize"] = able_to_finalize
```
Comment (Mark-Spiers-kainos) [inline cases/views_helpers.py]: Is conflicting advice still displayed to the user to update?

File: templates/case/views/advice-view.html [hunk 3]
```diff
@@ -3,15 +3,15 @@
     {% if case.application.application_type.key == 'open_licence' %}
         {% include 'case/views/open-goods.html' %}
     {% endif %}
-    {% if case.application.application_type.key == 'standard_licence' and not status_is_terminal %}
+    {% if case.application.application_type.key == 'standard_licence' or case.application.application_type.key == 'exhibition_clearance' and not status_is_terminal %}
```
Comment (Mark-Spiers-kainos) [inline templates/case/views/advice-view.html]: will this expand with future clearance type applications? If so might be good to make a filter that allows all clearance types if this is expected behaviour

File: cases/views/advice.py (hunk 1 above)
Comment (stephanieboyd) [reply cases/views/advice.py]: Good spot, changed. Check the logic on it please

File: templates/case/views/advice-view.html (hunk 3 above)
Comment (stephanieboyd) [reply templates/case/views/advice-view.html]: We can worry about that when it comes to it

File: cases/views_helpers.py [hunk 4]
```diff
@@ -63,7 +56,36 @@ def get_case_advice(get_advice, request, case, user_team_final, team=None):
     context["status_is_read_only"] = status_props["is_read_only"]
     context["status_is_terminal"] = status_props["is_terminal"]
 
-    return render(request, "case/advice/" + user_team_final + ".html", context)
+    return render(request, "case/advice/" + advice_level + ".html", context)
```
Comment (BenAndrew-Kainos) [inline cases/views_helpers.py]: f string please

File: cases/views_helpers.py [hunk 5]
```diff
@@ -63,7 +56,36 @@ def get_case_advice(get_advice, request, case, user_team_final, team=None):
...
+    elif case_type in CLEARANCE_CASE_TYPES and Permission.MANAGE_CLEARANCE_FINAL_ADVICE.value in permissions:
+        return True
+    else:
+        return False
+
+
+def _can_advice_be_finalised(advice):
+    """Check that there is no conflicting advice and that the advice can be finalised. """
+    for item in advice["advice"]:
+        if item["type"]["key"] == "conflicting":
```
Comment (BenAndrew-Kainos) [inline cases/views_helpers.py]: Could we store conflicting as a constant somewhere please

File: templates/case/views/advice-view.html [hunk 6]
```diff
@@ -3,15 +3,15 @@
...
 
     <!-- Destinations/End User + Ultimate End Users -->
     {% if case.application.application_type.key == 'open_licence' %}
         {% include 'case/views/open-destinations.html' %}
     {% endif %}
-    {% if case.application.application_type.key == 'standard_licence' %}
+    {% if case.application.application_type.key == 'standard_licence' or case.application.application_type.key == 'exhibition_clearance' %}
         {% include 'case/views/standard-destinations.html' %}
     {% endif %}
 </div>
```
Comment (BenAndrew-Kainos) [inline templates/case/views/advice-view.html]: newline

File: templates/case/views/advice-view.html [hunk 7]
```diff
@@ -3,15 +3,15 @@
...
+    {% if case.application.application_type.key == 'standard_licence' or case.application.application_type.key == 'exhibition_clearance' and not status_is_terminal %}
         {% include 'case/views/standard-goods.html' %}
     {% endif %}
 
     <!-- Destinations/End User + Ultimate End Users -->
     {% if case.application.application_type.key == 'open_licence' %}
         {% include 'case/views/open-destinations.html' %}
     {% endif %}
-    {% if case.application.application_type.key == 'standard_licence' %}
+    {% if case.application.application_type.key == 'standard_licence' or case.application.application_type.key == 'exhibition_clearance' %}
```
Comment (BenAndrew-Kainos) [inline templates/case/views/advice-view.html]: could you use in [standard,exhibition] (I'm fussy like that)

File: templates/case/advice/final.html [hunk 8]
```diff
@@ -21,41 +21,39 @@ <h1 class="govuk-heading-xl">
...
-                        {% else %}
-                            <a class="govuk-button" href="{% url 'cases:coalesce_team_advice' case.id %}" class="lite-tabs__tab" id="combine_advice_button">
-                                Combine all team advice
-                            </a>
-                        {% endif %}
-                    {% endif %}
-                {% endif %}
+            {% if not status_is_terminal and permitted_to_give_final_advice %}
+				{% if case.has_advice.team %}
+					{% if all_advice|length %}
```
Comment (BenAndrew-Kainos) [inline templates/case/advice/final.html]: Do you need length here (is it not just if all_advice)

File: cases/views_helpers.py (hunk 4 above)
Comment (stephanieboyd) [reply cases/views_helpers.py]: Changed

File: cases/views_helpers.py (hunk 5 above)
Comment (stephanieboyd) [reply cases/views_helpers.py]: Moved as discussed

File: templates/case/advice/final.html (hunk 8 above)
Comment (stephanieboyd) [reply templates/case/advice/final.html]: Not sure, it was existing. I just indented :)

File: templates/case/views/advice-view.html (hunk 6 above)
Comment (stephanieboyd) [reply templates/case/views/advice-view.html]: Added

File: templates/case/views/advice-view.html (hunk 7 above)
Comment (stephanieboyd) [reply templates/case/views/advice-view.html]: As discussed, won't change :)

**Pull request merged**